ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/wizard/v1/wizard/clientlibs/site/js/wizardview.js

```diff
@@ -301,11 +309,75 @@
                 this.#navigateAndFocusTab(lastVisibleIndex);
             }
+            this.#hideUnhideNavButtons(this.#_active);
+        }
+
+        /**
+         * Hides or shows, next and previous navigation buttons
+         *
+         * @private
+         * @param {Number} index The index of the tab to navigate to
+         * @param {Number} total number of tabs
+         */
+        #hideUnhideNavButtons(activeTabIndex) {
+            let tabsLength = this.#getCachedTabs() ? this.#getCachedTabs().length : 0;
+
+            let nextVisible = this.#findNextVisibleChildIndex(activeTabIndex);
+            let previousVisible = this.#findLastVisibleChildIndex(activeTabIndex);
+
+            if(tabsLength === 0 || this.maxEnabledTab <= this.minEnabledTab) {
+                this.getPreviousButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"false");
+                this.getNextButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"false");
+            }
+
+            if(activeTabIndex <= this.minEnabledTab || previousVisible === -1) {
+                this.getPreviousButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"false");
+            }
+            if(activeTabIndex === this.maxEnabledTab || nextVisible === -1) {
+                this.getNextButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"false");
+            }
+
+            if(tabsLength > 1 && activeTabIndex > this.minEnabledTab && previousVisible > -1) {
+                this.getPreviousButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"true");
+            }
+
+            if(activeTabIndex < this.maxEnabledTab && nextVisible > -1) {
+                this.getNextButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"true");
+            }
+        }
+
+        #setNavigationRange() {
+            let wizardPanels = this.#getCachedWizardPanels();
+            if(wizardPanels) {
+                this.maxEnabledTab = wizardPanels.length-1;
+                this.minEnabledTab = 0;
+                for (let i = 0; i < wizardPanels.length; i++) {
+                    if(!this.#childComponentEnabled(this.#getCachedWizardPanels()[i])) {
+                        this.minEnabledTab = i+1;
+                    } else {
+                        break;
+                    }
+                }
+                for (let i = wizardPanels.length - 1; i >= 0; i--) {
+                    if(!this.#childComponentEnabled(this.#getCachedWizardPanels()[i])) {
+                        this.maxEnabledTab = i;
+                    } else {
+                        break;
+                    }
+                }
+                this.minEnabledTab = Math.max(0, this.minEnabledTab);
+                this.maxEnabledTab = Math.min(this.#getCachedTabs().length-1, this.maxEnabledTab);
+            }
+        }
+
+        #childComponentEnabled(wizardTab) {
+            return wizardTab.children[0].getAttribute('data-cmp-visible') === 'true';
```

Comment:
This method name suggests that it checks whether child is enabled or not and it checks whether component is visible or not.
Please rectify this.